lib/std/x/os/socket.zig

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2021 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+
+const std = @import("../../std.zig");
+const net = @import("net.zig");
+
+const os = std.os;
+const fmt = std.fmt;
+const mem = std.mem;
+const time = std.time;
+const builtin = std.builtin;
+
+/// Import in a `Socket` abstraction depending on the platform we are compiling against.
+pub usingnamespace switch (builtin.os.tag) {
+    .windows => @import("socket_windows.zig"),
+    else => @import("socket_posix.zig"),
+};
+
+/// A common subset of shared structs across cross-platform abstractions over socket syscalls.
+pub fn Mixin(comptime Self: type) type {
```

Comment:
pedantic: maybe call it `ConnAddrMixin` ? especially since it is pub.
To me, calling it mixin is similar to calling a variable `object`.

Reply:
Hmm, I'll see about addressing this in a subsequent PR - maybe it might be better to just not forego mixins and define them as separate structs.

Reply:
Had to do a rebase just now, cleaned up how methods/structs are mixed in within the new commit.